CHANGELOG-1.4.md

```diff
@@ -0,0 +1,114 @@
+  * [v1.4.0](#v140)
+     * [Downloads for v1.4.0](#downloads-for-v140)
+        * [KubeEdge Binaries](#kubeedge-binaries)
+        * [Installer Binaries](#installer-binaries)
+        * [EdgeSite Binaries](#edgesite-binaries)
+     * [KubeEdge v1.4 Release Notes](#kubeedge-v14-release-notes)
+        * [1.4 What's New](#14-whats-new)
+        * [Other Notable Changes](#other-notable-changes)
+        * [Bug Fixes](#bug-fixes)
+  * [v1.4.0-beta.0](#v140-beta0)
+     * [Changelog since v1.3.1](#changelog-since-v131)
+        * [Bug Fixes](#bug-fixes-1)
+        
+# v1.4.0
+
+## Downloads for v1.4.0
+
+### KubeEdge Binaries
+| filename | Size | sha512 hash |
+| -------- | ---- | ----------- |
+
+
+### Installer Binaries
+| filename | Size | sha512 hash |
+| -------- | ---- | ----------- |
+
+### EdgeSite Binaries
+| filename | Size | sha512 hash |
+| -------- | ---- | ----------- |
+
+## KubeEdge v1.4 Release Notes
+
+### 1.4 What's New
+
+**Enhance Devices Management**
+
+Upgrade device API from v1alpha1 to v1alpha2, enhancement include: 
+
+- A new field is added to explicitly provide customized protocol support
+- Introduced `data` section to allow users defining data to get and process on the edge
+- Moved `propertyVistors` from device model to Device instance API
+
+Users are now able to customize the protocol of the edge device and also able to get and process the date in edge side.([#1687](https://github.com/kubeedge/kubeedge/pull/1687), [@luogangyi](https://github.com/luogangyi), [@kevin-wangzefeng](https://github.com/kevin-wangzefeng)).
+
+**Metrics-Server Support for metrics collection across cloud and edge**
+
+Users are now able to deploy metrics-server to collect resource metrics from edge nodes. Follow the [instructions here](/docs/setup/keadm.md#support-metrics-server-in-cloud) to deploy the metrics-server. ([#1735](https://github.com/kubeedge/kubeedge/pull/1735), [@Poor12](https://github.com/Poor12), [@kadisi](https://github.com/kadisi)).
+
+**EdgeNode Certificate Rotation**
+
+EdgeNodes are now able to automatically apply for new certificate when the certificate is about to expire and enforce TLS between CloudCore and EdgeCore. ([#1838](https://github.com/kubeedge/kubeedge/pull/1838), [@XJangel](https://github.com/XJangel), [@fisherxu](https://github.com/fisherxu))
+
+**Kubernetes Dependencies Upgrade**
+
+Upgrade the venderod kubernetes version to v1.18.6, users now can use the feature of new version
+on the cloud and on the edge side. ([#1982](https://github.com/kubeedge/kubeedge/pull/1982), [@dingyin](https://github.com/dingyin))
+
+
```

Comment:
```suggestion
### Important Steps before Upgrading

**Device CRD API change**
KubeEdge Device Controller now only reflects v1alpha2 for Device CRD, please make sure [Device v1alpha2](https://github.com/kubeedge/kubeedge/blob/release-1.4/build/crds/devices/devices_v1alpha2_device.yaml) and [DeviceModel v1alpha2](https://github.com/kubeedge/kubeedge/blob/release-1.4/build/crds/devices/devices_v1alpha2_devicemodel.yaml) installed, and you would need to manually convert their existing custom resources from v1alpha1 to v1alpha2.

It's recommended to keep v1alpha1 CRD and custom resources in the cluster or exported somewhere before upgrading, in case any rollback is needed.

```